### DISCUSSION TITLE
[Bug] `prefers-color-scheme` in SVG does not work for mobile dark mode

### DISCUSSION THREAD

**Discussion by LitoMore** [community/community#62430] [Mobile]
### Select Topic Area

Bug

### How to reproduce

See https://github.com/simple-icons/simple-icons. The readme of the project uses some SVG with `prefers-color-scheme` inside.

1. Use GitHub mobile app open https://github.com/simple-icons/simple-icons
2. Switch your mobile system color theme to dark mode
3. The logo color should be white, but the current behavior is black

This issue also can be reproduced in the discussion.

See the picture below:

```html
<img width="200" src="https://cdn.simpleicons.org/github/000/fff" />
```

<img width="200" src="https://cdn.simpleicons.org/github/000/fff" />

### References

- https://github.com/LitoMore/simple-icons-cdn
- [CSS `prefers-color-scheme`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme)

### Similar topic

We should consider some edge scenarios when implementing this feature. Here is a similar topic:

- https://github.com/orgs/community/discussions/61789

**Comment by mxie** [comment]
Thanks for reporting this @LitoMore! This is currently a known issue on our end that we don't have a fix for just yet. Can you tell me what device you're on?

**Reply by LitoMore** [reply]
iOS.

**Comment by ShineThyuZan** [comment]
Use this `"<img width="200" src="https://cdn.simpleicons.org/github" />"` instead of the original. (**REMOVE "/000/fff"**)

**Reply by LitoMore** [reply]
Hi @ShineThyuZan, thank you for helping. But please kindly read through my description about this issue.

The current problem is the GitHub mobile app does not support render a dark mode color that `prefers-color-scheme` which inside of the SVG.

I know there is a solution to use `<picture>` instead. But the topic in this discussion is I hope they can support `perfers-color-scheme` inside of the SVG.

**Comment by nicolo-ribaudo** [comment]
Hi! 2 years later I just got a report of this same problem, with the image at https://github.com/tc39/agendas/blob/main/2025/05.md after "Total duration of scheduled discussions" (https://camo.githubusercontent.com/b5b499ef059be6ff380d629616176cfa00d12fc3838ad6dbd94651b5f08cf6c2/68747470733a2f2f746333392d6167656e64612d74696d652e64656e6f2e6465762f323032352f30352f)

**Comment by iskae** [comment]
Hey folks 👋 

As far as I can see this issue was fixed on iOS side. Can you please confirm @LitoMore ? 

Are you also using iOS @nicolo-ribaudo ? 

Thanks in advance!

**Reply by nicolo-ribaudo** [reply]
I was using iOS. If you have dark mode enabled and you are seeing the time after "Total duration of scheduled discussions" as white test, then it's indeed working for you. If you are seeing it as black text with a white outline (like a neon effect), then you are seeing the light mode image and not the dark mode one.

**Reply by nicolo-ribaudo** [reply]
Bad: 
<img width="117" alt="image" src="https://github.com/user-attachments/assets/a3e50b6f-93ba-4643-be42-3b41f624e1c3" />

Good: 
<img width="117" alt="image" src="https://github.com/user-attachments/assets/55c775e3-bb57-49f5-bea8-b59be411db7b" />

**Reply by LitoMore** [reply]
I'm not sure it can be fixed from GitHub app side since it's a WebKit issue: https://bugs.webkit.org/show_bug.cgi?id=199134

**Reply by iskae** [reply]
Just checked in with our iOS colleagues and looks like @LitoMore is correct ❤️ Unfortunately we can't do much until the webkit bug is resolved 😞 On our side we are handling `prefers-color-scheme` correctly as mentioned above 🙇

**Reply by nicolo-ribaudo** [reply]
Uh, I can confirm, I can also reproduce it in a desktop WebKit-browser browser :/

Sorry for assuming it was a bug on GitHub's side, I'll go ask WebKit people about it instead :stuck_out_tongue:

**Reply by iskae** [reply]
Oh, nothing to be sorry about at all ❤️ Please keep the bug reports / suggestions coming in 🙇 Always happy to hear from our users 😊
